src/utils/searchViewHelper.js

```diff
@@ -32,24 +47,35 @@ const searchViewHelper = {
     // If repository is selected after search value is input, fetch search results
     if (search !== '') {
       searchViewHelper.searchByRevisionOrEmail(repository, search);
+    } else {
+      // Fetch 10 most recent revisions when repository changes
+      store.dispatch(fetchRecentRevisions(repository));
     }
-    // Fetch 10 most recent revisions when repository changes
-    store.dispatch(fetchRecentRevisions(repository));
   },
 
   handleChangeSearch(event) {
     const search = event.target.value;
     const { repository } = store.getState().search;
     store.dispatch(updateSearchValue(search));
+    store.dispatch(updateSearchResults([]));
+    store.dispatch(updateInputError(false));
+    store.dispatch(updateInputHelperText(''));
+
+    const _idleTime = 500;
+    const _onTimeout = () => {
```

Comment:
(optional) I'm somewhat surprised that you're using the prefix `_` for these 2 variables. Not that I'm opposed, but I was wondering if you were following some conventions for this.